internal/otelcollector/config/metric/agent/receivers.go

```diff
@@ -138,3 +142,66 @@ func makePrometheusAppPodsConfig() *PrometheusReceiver {
 		},
 	}
 }
+
+func makePrometheusIstioConfig() *PrometheusReceiver {
+	return &PrometheusReceiver{
+		Config: promconfig.Config{
+			ScrapeConfigs: []*promconfig.ScrapeConfig{
+				{
+					JobName:     "istio-proxy",
+					MetricsPath: "/stats/prometheus",
+					ServiceDiscoveryConfigs: []promdiscovery.Config{
+						&promk8sdiscovery.SDConfig{
+							Role:             promk8sdiscovery.RoleEndpoint,
+							HTTPClientConfig: promcommonconfig.DefaultHTTPClientConfig,
+						},
+					},
+					RelabelConfigs: []*promlabel.Config{
+						{
+							SourceLabels: []prommodel.LabelName{"__meta_kubernetes_pod_node_name"},
+							Regex:        promlabel.MustNewRegexp(fmt.Sprintf("$%s", config.EnvVarCurrentNodeName)),
+							Action:       promlabel.Keep,
+						},
+						{
+							SourceLabels: []prommodel.LabelName{"__meta_kubernetes_pod_container_name"},
+							Action:       promlabel.Keep,
+							Regex:        promlabel.MustNewRegexp("istio-proxy"),
+						},
+						{
+							SourceLabels: []prommodel.LabelName{"__meta_kubernetes_pod_container_port_name"},
+							Action:       promlabel.Keep,
+							Regex:        promlabel.MustNewRegexp("http-envoy-prom"),
+						},
+						{
+							SourceLabels: []prommodel.LabelName{"__meta_kubernetes_namespace"},
+							Action:       promlabel.Replace,
+							TargetLabel:  "namespace",
+						},
+						{
+							SourceLabels: []prommodel.LabelName{"__meta_kubernetes_service_name"},
+							Action:       promlabel.Replace,
+							TargetLabel:  "service",
+						},
+						{
+							SourceLabels: []prommodel.LabelName{"__meta_kubernetes_pod_name"},
+							Action:       promlabel.Replace,
+							TargetLabel:  "pod",
+						},
+						{
+							SourceLabels: []prommodel.LabelName{"__meta_kubernetes_pod_node_name"},
+							Action:       promlabel.Replace,
+							TargetLabel:  "node",
+						},
```

Comment:
Can you also check in what state a pod should be in order to be scrapeable, see __meta_kubernetes_pod_ready and __meta_kubernetes_pod_phase labels https://prometheus.io/docs/prometheus/latest/configuration/configuration/#pod

I haven't considered that yet, might be that we explicitly need to check that it is not in "Completed" state?

Reply:
Seeing this one here: https://raw.githubusercontent.com/istio/istio/release-1.8/samples/addons/prometheus.yaml
It looks like we should add something like:
```
     - action: drop
        regex: Pending|Succeeded|Failed
        source_labels:
        - __meta_kubernetes_pod_phase
```
Please check that also for the annotation based pod scraping, same situation

Reply:
pod phase check rule added